src/dataregistry/registrar/dataset.py

```diff
@@ -924,112 +935,28 @@ def _delete_by_id(self, dataset_id, confirm=False):
-            # Make sure keyword is a string
-            if not isinstance(k, str):
-                raise ValueError(f"{k} is not a valid keyword string")
-
-        # Make sure keywords are all in the keywords table
-        keyword_table = self._get_table_metadata("keyword")
-
-        stmt = select(keyword_table.c.keyword_id).where(
-            keyword_table.c.keyword.in_([x.lower() for x in keywords])
-        )
-
-        with self._engine.connect() as conn:
-            result = conn.execute(stmt)
-            conn.commit()
-
-        # Keyword found
-        for r in result:
-            keyword_ids.append(r.keyword_id)
-
-        # Keyword not found
-        if len(keyword_ids) != len(keywords):
-            raise ValueError("Not all keywords selected are registered")
-
-        return keyword_ids
-
-    def add_keywords(self, dataset_id, keywords):
-        """
-        Add/append keywords to an already existing dataset.
-
-        First check the keywords are valid, then append. If the dataset already
-        has one or more of the passed keywords attributed to it, the keyword(s)
-        will not be duplicated.
+        Add keyword tags to a dataset entry.
 
         Parameters
         ----------
         dataset_id : int
-        keywords : list[str]
+            Dataset to add keyword to
+        keyword : list[str]
+            Keyword to add to dataset
         """
+        self.keyword_table.add_dataset_keywords_relation(dataset_id, keyword)
 
-        # Make sure things are valid
-        if not isinstance(keywords, list):
-            raise ValueError("Passed keywords object must be a list")
-
-        if len(keywords) == 0:
-            return
-
-        # Validate keywords (make sure they are in the `keyword` table)
-        keyword_ids = self._validate_keywords(keywords)
-
-        # Link fo the dataset-keyword association table
-        dataset_keyword_table = self._get_table_metadata("dataset_keyword")
-
-        with self._engine.connect() as conn:
-            # Loop over each keyword in the list
-            for keyword_id in keyword_ids:
-                # Check if this dataset already has this keyword
-                stmt = (
-                    select(dataset_keyword_table)
-                    .where(dataset_keyword_table.c.dataset_id == dataset_id)
-                    .where(dataset_keyword_table.c.keyword_id == keyword_id)
-                )
-
-                result = conn.execute(stmt)
-                rows = result.fetchall()
-
-                # If we don't have the keyword, add it
-                if len(rows) == 0:
-                    add_table_row(
-                        conn,
-                        dataset_keyword_table,
-                        {"dataset_id": dataset_id, "keyword_id": keyword_id},
-                        commit=False,
-                    )
-
-            conn.commit()
-
-    def delete_keywords(self, dataset_id, keywords):
+    def delete_keywords(self, dataset_id, keyword):
```

Comment:
I'd prefer to rename to "remove_keywords".   It's a change in interface, but I'm sure no actual user has actually called this routine.   It shouldn't be disruptive.